compiler/rustc_typeck/src/collect.rs

```diff
@@ -2683,6 +2683,14 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, id: DefId) -> CodegenFnAttrs {
         codegen_fn_attrs.flags |= CodegenFnAttrFlags::TRACK_CALLER;
     }
 
+    // With -Z panic-in-drop=abort, drop_in_place never unwinds.
+    if tcx.sess.opts.debugging_opts.panic_in_drop == PanicStrategy::Abort {
+        let drop_in_place = tcx.require_lang_item(LangItem::DropInPlace, None);
+        if id == drop_in_place {
+            codegen_fn_attrs.flags |= CodegenFnAttrFlags::NEVER_UNWIND;
```

Comment:
It's undefined behaviour if a `nounwind` function actually unwinds.

Reply:
Oh actually this attribute is already covered by `abort_unwinding_calls` transform. So it won't be UB.